azure/converters/image.go

```diff
@@ -57,7 +57,7 @@ func computeImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
 	idTemplate := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s"
 	if image.SharedGallery != nil {
 		return &compute.ImageReference{
-			ID: pointer.String(fmt.Sprintf(idTemplate,
+			SharedGalleryImageID: pointer.String(fmt.Sprintf(idTemplate,
```

Comment:
Just to clarify, would we still want to set the `ID` field for non-shared images? In other words, would we want to preserve the existing functionality?

Reply:
I'm unsure on this one. I don't have a way that I know of to exercise this code to see that is returned from the Azure API, and since the `SharedGalleryImageID` field existed it seemed to make sense to me to use it, but maybe that was a bad idea since it originally used ID.

Reply:
I switched back to ID.